tests/sparseml/transformers/test_transformers.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import math
+import os
+import shutil
+import tarfile
+
+import numpy as np
+import onnx
+import onnxruntime as ort
+import pytest
+from transformers import AutoConfig
+
+from sparseml.transformers.sparsification import Trainer
+from sparsezoo import Zoo
+from src.sparseml.transformers import export_transformer_to_onnx, load_task_model
+
+
+def _is_yaml_recipe_present(model_path):
+    return any(
+        [
+            file
+            for file in glob.glob(os.path.join(model_path, "*"))
+            if (file.endswith(".yaml") or ("recipe" in file))
+        ]
+    )
+
+
+def _run_inference_onnx(path_onnx, input):
+    ort_sess = ort.InferenceSession(path_onnx)
```

Comment:
let's refactor this function so we do not hardcode stuff such as number of inputs and number of input names. (number of inputs can just come from unpacking the data point). input names can come from the onnx graph.

```python

model = onnx.load(path_onnx)
input_names = [inp.name for inp in model.graph.input]
```